### PR TITLE
Wendell branch

### PR DESCRIPTION
**Pull request by PsYear**
use tf.dataset to be the input of Keras

### CODE REVIEW

**Comment by PsYear** [comment]
miss validation and test

**Comment by PsYear** [comment]
@ytdu  @Scenny

File: lm_keras/main.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    k_model,ppl = train_model.build_lm_model(itera_x_train, itera_y_train)
+    valid_loss = k_model.evaluate(x=itera_x_valid, y=itera_y_valid,steps=m_config.valid_steps)[0]
+    test_loss = k_model.evaluate(x=itera_x_test, y=itera_y_test,steps=m_config.test_steps)[0]
+    print("trainning's ppl:", ppl)
+    print("valid's ppl:", math.exp(valid_loss))
+    print("test's ppl:", math.exp(test_loss))
+
+
+if __name__ == "__main__":
+    main()
```
Comment (ytdu) [inline lm_keras/main.py]: `main.py` should support argument parsing (package `argparse` is recommended), model training, model saving, and model loading and testing. Perhaps loading model and generating random sentences in addition.

File: lm_keras/model.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+from keras.optimizers import SGD
+from keras.models import Model
+from keras.layers import Input, Dense
+from keras import backend as k
+from config import m_config
+
+
+class LanguageModel(object):
+    def __init__(self):
+        self.config = m_config
```
Comment (ytdu) [inline lm_keras/model.py]: pass config as an argument, to decouple the definition of config and the model.

File: lm_keras/model.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        x = LSTM(
+            m_config.units,
+            activation='tanh',
+            recurrent_activation='hard_sigmoid',
+            use_bias=True,
+            kernel_initializer='glorot_uniform',
+            recurrent_initializer='orthogonal',
+            bias_initializer='zeros',
+            return_sequences=True)(x)
+        x = LSTM(
```
Comment (ytdu) [inline lm_keras/model.py]: use a config to control num layers of RNN.

File: lm_keras/model.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            nesterov=False)
+        model = Model(inputs=inputs, outputs=predictions)
+        model.compile(
+            loss='categorical_crossentropy',
+            optimizer=sgd,
+            metrics=['accuracy'],
+            target_tensors=[
+                targets.get_next()])
+        print(model.summary())
+        history = model.fit(
```
Comment (ytdu) [inline lm_keras/model.py]: Fitting model should be put here or outside?

**Pull request merged**